### ISSUE TITLE
Update autograph calls to send the right parameters for recommended extensions

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-server#11062]
Autograph is going to be updated to provide support for adding to the signature of an extension so that Firefox will know that a curated extension is going to be used.

Once that support is ready and we have already done the backend work to support the identification of recommended add-ons we can then conditionally send different params to autograph to request the correct signature attribution.

https://bugzilla.mozilla.org/show_bug.cgi?id=1548647

**Comment by EnTeQuAk** [comment]
QA: If an add-on is marked as being pending recommendation and a new version gets uploaded so that it lands in the recommendation queue, that new version will have added a new `mozilla-recommendation.json` file that has been generated by autograph.

**Comment by AlexandraMoga** [comment]
@EnTeQuAk I've run a test to see if this works, but at the end there was no `mozilla-recommendation.json` file present in the recommended version.

These are the exact steps that I've taken:
1. Uploaded a new extension
2. Added the extension to the recommended list in Discovery items => add-on state is "pending recommendation"
3. Uploaded a new version for this extension => new version lands in the Recommended review queue
4. Approve the new version 
5. Verify the add-on package after approval to see if the `mozilla-recommendation.json` file is present => no such file was created

This is the add-on used in the described flow:
https://addons-dev.allizom.org/en-US/firefox/addon/group-speed-dials/
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/group-speed-dials

**Comment by diox** [comment]
Tested on stage and it's not working either. See https://reviewers.addons.allizom.org/en-US/reviewers/review/laping-chingood-weaver

**Comment by AlexandraMoga** [comment]
@EnTeQuAk  I've reverified this flow on -dev and stage and the results are looking fine.

I can confirm that:

- when an add-on is approved from the Recommended queue, after signing, the package receives the `mozilla-recommendation.json` file (see screenshot)
   -  some examples from -dev and stage: 
     https://addons-dev.allizom.org/en-US/firefox/addon/auto-tab-discard/
     https://addons-dev.allizom.org/en-US/firefox/addon/greek-spell-dictionary/
     https://addons.allizom.org/en-US/firefox/addon/git-history-browser-extension/
- if the add-on is removed from the recommended list in Discovery items, the next version submitted will be auto-approved and the `mozilla-recommendation.json` file is no longer present in the signed package
- users who had an add-on installed before it was recommended will receive an update to the new recommended version
- If an add-on in `pending recommendation` state receives a new unlisted version, that version will also be signed with the `mozilla-recommendation.json` file


![image](https://user-images.githubusercontent.com/31961530/59421340-1ae1c700-8dd7-11e9-93f6-9e16b85a9343.png)
